jobrunrpro/src/main/java/org/axonframework/extensions/jobrunrpro/scheduling/JobRunrProEventScheduler.java

```diff
@@ -0,0 +1,232 @@
+     */
+    public void cancelAll() {
+        deleteAllPendingJobsByLabel(jobScheduler, storageProvider, getLabel(jobName), DELETE_REASON);
+    }
+
+    /**
+     * Cancels all the events scheduled with the given {@code formerJobName}. This method has no impact on events which
+     * have already been triggered.
+     *
+     * @param formerJobName a {@link String} with a name used before
+     */
+    public void cancelAll(String formerJobName) {
+        deleteAllPendingJobsByLabel(jobScheduler, storageProvider, getLabel(formerJobName), DELETE_REASON);
+    }
+
+    /**
+     * Builder class to instantiate a {@link JobRunrProEventScheduler}.
+     * <p>
+     * The {@link TransactionManager} is defaulted to a {@link NoTransactionManager}.
+     * <p>
+     * The {@link JobScheduler}, {@link StorageProvider}, {@link Serializer} and {@link EventBus} are <b>hard
+     * requirements</b> and as such should be provided.
+     */
+    public static class Builder {
+
+        private JobScheduler jobScheduler;
+        private StorageProvider storageProvider;
+
+        private String jobName = "AxonScheduledEvent";
+        private Serializer serializer;
+        private TransactionManager transactionManager = NoTransactionManager.INSTANCE;
+        private EventBus eventBus;
+
+        /**
+         * Sets the {@link JobScheduler} used for scheduling and triggering purposes of the events.
+         *
+         * @param jobScheduler a {@link JobScheduler} used for scheduling and triggering purposes of the events
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder jobScheduler(JobScheduler jobScheduler) {
+            assertNonNull(jobScheduler, "JobScheduler may not be null");
+            this.jobScheduler = jobScheduler;
+            return this;
+        }
+
+        /**
+         * Sets the {@link StorageProvider} used for the cancel all methods. Those will return an error when called
+         * while the storage provider has not been set.
+         *
+         * @param storageProvider a {@link StorageProvider} used for scheduling and triggering purposes of the
+         *                        deadlines
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder storageProvider(StorageProvider storageProvider) {
+            assertNonNull(storageProvider, "StorageProvider may not be null");
+            this.storageProvider = storageProvider;
+            return this;
+        }
+
+        /**
+         * Sets the {@link String} used for creating the scheduled jobs in JobRunr. Defaults to
+         * {@code AxonScheduledEvent}.
+         *
+         * @param jobName a {@link JobScheduler} used for naming the job
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder jobName(String jobName) {
+            assertNonNull(jobName, "JobName may not be null");
```

Comment:
I should've reacted on this in the AF PR too...but I think `BuilderUtils#assertNonEmpty` would be better suited for `String` validation in the builder.

Reply:
No longer relevant, with the JobRuneEventScheduler being removed.